server/protocols/webadmin/webadmin-mailbox/src/main/java/org/apache/james/webadmin/routes/UserQuotaRoutes.java

```diff
@@ -99,17 +101,22 @@ private static Optional<Integer> intQueryParameter(Request request) {
             }
         }
 
-        private static Optional<QuotaComponent> getQuotaComponent(Request request) {
-            String quotaComponentString = request.queryParams(USERS_PER_SECOND);
-            if (Objects.isNull(quotaComponentString)) {
-                return Optional.empty();
+        private static List<QuotaComponent> getQuotaComponent(Request request) {
+            List<QuotaComponent> quotaComponents = new ArrayList<>();
+            String[] quotaComponentStrings = request.queryParamsValues(QUOTA_COMPONENT);
```

Comment:
Shouldn't you be able to work on a stream directly from `request.queryParamsValues(QUOTA_COMPONENT)`?

Reply:
Because I have to throw exception if any quotaComponentString is wrong, it is difficult to use stream

Reply:
How about something like this:

```java
            return Arrays.stream(quotaComponentStrings)
                .map(quotaComponentString -> {
                    QuotaComponent quotaComponent = QUOTA_COMPONENT_MAP.get(quotaComponentString);
                    if (Objects.isNull(quotaComponent)) {
                        throw new IllegalArgumentException(String.format("Illegal value supplied for query parameter '%s' with value '%s', expecting existing " +
                            "quota components", QUOTA_COMPONENT, quotaComponentString));
                    }
                    return quotaComponent;
                })
                .collect(Collectors.toUnmodifiableList());
```

Reply:
oh tks don't know why I thought this way did not work